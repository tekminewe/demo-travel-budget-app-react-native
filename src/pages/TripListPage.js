// @flow
import React from 'react';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import EmptyTrip from '../components/EmptyTrip';
import { type TripStateType } from '../reducers/TripReducer';
import TripList from '../components/TripList';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

type PropsType = {
  navigation: any,
  trips: $PropertyType<TripStateType, 'trips'>,
}

export const TripListPage = ({ navigation, trips }: PropsType) => {

  if (!trips.length) {
    return (
      <EmptyTrip onAdd={() => navigation.push('CreateTrip')} />
    )
  }

  return (<TripList trips={trips} />);
};

TripListPage.navigationOptions = {
  title: 'Trips',
}

const mapStateToProps = (state) => ({
  trips: state.trip.trips,
});

export default connect(mapStateToProps, null)(TripListPage);
