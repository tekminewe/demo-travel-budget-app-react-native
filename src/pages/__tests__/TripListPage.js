import React from 'react';
import renderer from 'react-test-renderer';
import TripListPage from '../TripListPage';

describe('TripListPage', () => {
  test('should display add button if no trips', () => {
    const page = renderer.create(<TripListPage />).toJSON();
    expect(page).toMatchSnapshot();
  });
});
