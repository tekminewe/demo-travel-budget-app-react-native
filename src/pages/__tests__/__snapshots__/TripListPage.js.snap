// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TripListPage should display add button if no trips 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    You do not have any trip yet.
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add Trip
      </Text>
    </View>
  </View>
</View>
`;

exports[`TripListPage should trips list if have trips 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "name": "Trip 1",
        },
        Object {
          "id": 2,
          "name": "Trip 2",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Id: 1
          </Text>
          <Text>
            Name: Trip 1
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            Id: 2
          </Text>
          <Text>
            Name: Trip 2
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
